README.md

```diff
@@ -36,339 +34,172 @@
-  patterns:
-    - pattern: |
-        public $RETURN $METHOD(...){
-            ...
-            make_transaction($T);
-            ...
-        }
-    - pattern-not: |
-        public $RETURN $METHOD(...){
-            ...
-            verify_transaction($T);
-            ...
-            make_transaction($T);
-            ...
-        }
-  message: |
-    In $METHOD, there's a call to make_transaction() without first calling verify_transaction() on the Transaction object.
-```
-
-* `$RETURN`, `$METHOD`, and `$T` are *metavariables*, an abstraction that Semgrep provides when you want to match something but you don't know exactly what it is ahead of time.
-  * You can think of *metavariables* like a [capture group](https://regexone.com/lesson/capturing_groups) in regular expressions.
-* The `pattern` clause defines what we're looking for: any method that calls `make_transaction()`.
-* The `pattern-not` clause *filters out* matches we don't want; in this case, methods where a transaction (`$T`) is passed to `verify_transaction()` before `make_transaction()`.
-* The `message` is what's returned in Semgrep output, either to STDOUT or as a comment on the pull request on GitHub or other systems.
-  * Note that *metavariables* can be used to customize messages and make them
-    contextually relevant. Here we're helpfully telling the user the method
-    where we've identified the bug.
-
-You can play with this transaction example here: https://semgrep.dev/4b4g.
-
-**Learn More**
-
-* See the [pattern features docs](docs/pattern-features.md) for more info and
-  examples on the flexibility and power of Semgrep patterns.
-* See the [YAML configuration file docs](docs/configuration-files.md) for
-  details on all of the keys that can be used and how they work.
-* See [Semgrep rule writing methodology](docs/writing_rules/README.md) for
-  details on the thought process behind writing and rolling out a new Semgrep
-  rule successfully, and see [these docs](docs/writing_rules/examples.md) for
-  walkthroughs of writing rules from scratch.
-* There's also an interactive, example-based Semgrep rule writing tutorial here:
-  https://semgrep.dev/learn.
-
-### Run Semgrep Continuously in CI
-
-Semgrep can be run via CLI or Docker and output results as JSON (via the `--json` flag), so it can be inserted into any CI pipeline and have its results processed by whatever tools you're using.
-
-Semgrep is aware of *diffs*, so it can report only findings that occur in newly added code, for example, in a commit or pull request.
-
-Currently, the easiest way to integrate Semgrep into CI is via a GitHub action we've built. See the [integrations docs](docs/integrations.md) for more details.
-
-Semgrep can also output results in the standardized Static Analysis Results Interchange Format ([SARIF](https://docs.oasis-open.org/sarif/sarif/v2.1.0/cs01/sarif-v2.1.0-cs01.html)) with the `--sarif` flag, if you use tools that accept this format.
-
-## Upgrading
```

Comment:
We need to keep the `Upgrading` header - it's referenced by the upgrade CLI message:

> "A new version of Semgrep is available. Please see https://github.com/returntocorp/semgrep#upgrading for more information."

That or we need to change the upgrade message and put this information in a new location. But then of course older Semgrep versions will be pointing to the wrong location.

Reply:
Great catch. Added back, mind taking a look?